### PR TITLE
Code modernization

### PR DESCRIPTION
**Pull request by offa**
Some code modernization.

-------------------

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by timja — APPROVED** [review]
nice!

File: src/test/java/org/jenkinsci/remoting/protocol/ProtocolStackImplTest.java [hunk 1]
```diff
@@ -141,11 +141,8 @@ public void pipeCloseSource() throws Exception {
         pipe.source().close();
         assertThat(pipe.source().isOpen(), is(false));
         assertThat(pipe.sink().isOpen(), is(true));
-        try {
-            pipe.sink().write(ByteBuffer.allocate(1));
-        } catch (IOException e) {
-            assertThat(e.getMessage(), containsString("Broken pipe"));
-        }
+        final IOException e = assertThrows(IOException.class, () -> pipe.sink().write(ByteBuffer.allocate(1)));
```
Comment (offa) [inline src/test/java/org/jenkinsci/remoting/protocol/ProtocolStackImplTest.java:144]: This change breaks the Windows build somehow; Linux works here though.

Comment (basil) [reply src/test/java/org/jenkinsci/remoting/protocol/ProtocolStackImplTest.java]: > This change breaks

Therefore reverted in commit c336c910713a1b981cb0439beacbe5892dd5109a.

**Review by basil — APPROVED** [review]

File: src/test/java/org/jenkinsci/remoting/protocol/ProtocolStackImplTest.java (hunk 1 above)
Comment (offa) [reply src/test/java/org/jenkinsci/remoting/protocol/ProtocolStackImplTest.java]: Thanks!

**Pull request merged**